service-workers/service-worker/request-end-to-end.https.html

```diff
@@ -22,22 +22,34 @@
 
     function onRegister(registration) {
         var sw = registration.installing;
+        sw.addEventListener('statechange', t.step_func(function(event) {
+            if (event.target.state == 'activated') {
+                onActive(sw);
+            }
+        }));
+    }
+
+    function onActive(sw) {
         var port = sendMessagePort(sw);
         port.addEventListener('message', t.step_func(function(event) {
             onMessage(event);
         }), false);
         port.start();
-        sw.addEventListener('statechange', t.step_func(function(event) {
-            if (event.target.state == 'activated')
-                onActive();
-        }));
     }
 
-    function onActive() {
+    function onMessage(event) {
+        if (event.data === 'received port') {
+            onPortReady();
+        } else {
+            onResult(event);
+        }
+    }
+
+    function onPortReady() {
         with_iframe(scope).then(function(f) { frames.push(f); });
     }
 
-    function onMessage(event) {
+    function onResult(event) {
```

Comment:
unrelated to this change, but at some level the call to onResult needs to be wrapped in t.step/t.step_func to make sure the asserts get attributed to the correct test (pretty much all the other async code also needs to be wrapped to properly attribute exceptions of course). Or alternatively since the entire file only has one async test anyway, we could just get rid of the async_test completely turning this into a single page test (but some of the helper methods like service_worker_unregister_and_done wouldn't work anymore then)